daemon/cmd/daemon_main.go

```diff
@@ -1585,6 +1585,9 @@ var daemonCell = cell.Module(
 	"daemon",
 	"Legacy Daemon",
 
+	// Processes endpoint deletions that occurred while the agent was down.
```

Comment:
suggestion: add a comment that this must be started before the API is started. (what guarantees that? just the cell's ordering?)

Reply:
As long as we unlock after the socket file is created, that will work fine. The previous code (attempts to) do something like that.

Once the socket file exists, the CNI plugin will react accordingly.

Reply:
Done refactoring it. Added a separate start hook that depends on both API server and the deletion queue processing.